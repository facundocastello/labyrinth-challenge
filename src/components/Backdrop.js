import React from 'react';
import styled from 'styled-components';

const Relative = styled.div`
  position: relative;
  width: 0;
  height: 0;
`;
const Backdrop = styled.div`
  align-items: center;
  display:flex;
  flex-direction: column;
  background-color: rgba(0,0,0,.5);
  justify-content: center;
  height: ${( { size } ) => `${( size * 32 )}px`};
  width: ${( { size } ) => `${( size * 32 )}px`};
`;

export default ( { children, size } ) => {
  return <Relative data-testid="backdrop">
    <Backdrop size={size}>
      {children}
    </Backdrop>
  </Relative>;
};
