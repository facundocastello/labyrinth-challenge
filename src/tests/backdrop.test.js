import React from 'react';
import Backdrop from '../components/Backdrop';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe( 'Backdrop Block', () => {
  it( 'renders Backdrop', () => {
    const tree = renderer.create(
      <Backdrop />
    ).toJSON();
    expect( tree ).toMatchSnapshot();
  } );
} );