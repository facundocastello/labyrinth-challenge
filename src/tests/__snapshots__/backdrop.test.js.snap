// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop Block renders Backdrop 1`] = `
.c0 {
  position: relative;
  width: 0;
  height: 0;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: rgba(0,0,0,.5);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: NaNpx;
  width: NaNpx;
}

<div
  className="c0"
  data-testid="backdrop"
>
  <div
    className="c1"
  />
</div>
`;
